### DISCUSSION TITLE
Step length calibration

### DISCUSSION THREAD

**Discussion by haecksenwerk** [InfiniTimeOrg/InfiniTime#1624] [Ideas]
### Verification

- [X] I searched for similar feature request and found none was relevant.

### Pitch us your idea!

Calibrate step-length with Gadgetbridge to show an accurate distance

### Description

When a constant step count for more like say one minute is determined by the watch, a calibration process could be started by sending a "workout request" to Gadgetbridge, which then returns GPS data every second, which then could be buffered for some 30 seconds until sending a "workout ended". 

From the buffered GPS data a step length could be calculated then, which could be utilized to display a relatively accurate distance on the watch.

[https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/OpenTracks-API](https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/OpenTracks-API)

Related issue: #883

**Comment by ITCactus** [comment]
from my understanding, the pinetime/infinitime only detects the steps, and does not care about your step length. it syncs only steps count, and not a distance.
the distance based on the step counts is calculated on the companion app side (Gadgetbridge in this case).
if you are not happy with default values, you may adjust those according to your body in "Settings → About you" in Gadgetbridge as described [here](https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/Activity-Sessions-List#:~:text=screens%20→%20Settings%20icon.-,Step%20length,-→%20this%20will%20be):

> Activity List dashboard
Make sure to adjust your personal settings and goals, available in menu: Settings → About you (or now also available via Settings → About you in the Charts screens → Settings icon.
Step length → this will be used to calculate distance from the number of Active steps.

**Comment by haecksenwerk** [comment]
@ITCactus 
This is how I also understand the current step detection and the adjustments that are offered by Gadetbridge.

But I'm more referring to benefiting from the phone's GPS and making use of it to automatically determine a sufficient accuracy of the step size, which can then be used to display the distance traveled directly on the watch. This is why I also mentioned issue #883

Once the value has been determined and saved, the watch does not necessarily need to be paired to display the distance.

It would also be nice if the calibration could be quickly started from the watch menu so that the step size could be determined again for a hike, which is certainly different than when you walk around in the house. And the value would certainly be significantly different again if you want to have as accurate as possible stepsize to check the covered track while jogging.
